bats/fb-test-katello.bats

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bats
+# vim: ft=sh:sw=2:et
+
+set -o pipefail
+
+load os_helper
+load foreman_helper
+
+setup() {
+  tSetOSVersion
+}
+
+@test "check web app is up" {
+  curl -sk "https://localhost$URL_PREFIX/users/login" | grep -q login-form
+}
+
+@test "wake up puppet agent" {
+  source ~/.bashrc
+  puppet agent -t -v
+}
+
+@test "check web app is still up" {
+  curl -sk "https://localhost$URL_PREFIX/users/login" | grep -q login-form
+}
+
+@test "check smart proxy is registered" {
+  count=$(hammer --csv proxy list | wc -l)
+  [ $count -gt 1 ]
+}
+
+@test "check host is registered" {
+  [ x$FOREMAN_VERSION = "x1.3" ] && skip "Only supported on 1.4+"
```

Comment:
So I did in my local branch, let me polish and push that before you continue review.